### ISSUE TITLE
upgrading to more recent JCEF

### ISSUE THREAD

**Issue by binary1230** [montoyo/mcef#30]
Hi, I'm attempting to upgrade to latest JCEF/CEF so I can get a more recent chromium version for some video streaming stuff I'm trying to do.

I was able to get things compiled and working decently for a first pass, but currently I am having an issue where CefClient::OnPaint() never seems to get called.   I realize without providing any additional context it would be hard to give advice, but, before I get too much deeper on the debugging, any obvious gotchas come to mind on why that could be happening?

Thanks
-Dom

**Comment by binary1230** [comment]
and, figured it out, turned out that with the new JCEF debug DLL it had some new checks to make sure we're on the right Thread, after restoring some of the init code to be on the original thread JCEF was using, it started working!

Here's a shot of it running Chromium 67 inside MC
https://github.com/binary1230/mcef/commit/3d26ca58dbe867a832cf2edf26d5a47cf0310359

Will keep testing this and probably open a PR soon for it

**Comment by binary1230** [comment]
And here it is working with WebDisplays unmodified:

![image](https://user-images.githubusercontent.com/5413064/41504699-87aaf8e0-71c5-11e8-8c31-8406891c7cfb.png)

**Comment by grillo78** [comment]
Can you play a video with any codec?

**Comment by montoyo** [comment]
Thanks @binary1230 that was actually some good work and I'm really sorry I couldn't find some time to work on MCEF earlier. Since this is quite old, I've updated MCEF to the latest JCEF version myself, but your PR contains a lot of interesting stuff that I'll definitely check out.

The latest JCEF version contained a few others annoying things (like, the backspace key was broken, and also rendering was laggy if the mouse wasn't moving) that I had to fix, but it seems to be working. Release should be ready soon.

Anyway... thanks a lot!
